src/main/groovy/net/minecraftforge/gradleutils/GradleUtilsExtension.groovy

```diff
@@ -46,7 +51,15 @@ class GradleUtilsExtension {
      * @return a version in the form {@code $tag.$offset}, e.g. 1.0.5
      */
     String getFilteredTagOffsetVersion(boolean prefix = false, String filter) {
-        return GradleUtils.getFilteredTagOffsetBranchVersion(getGitInfo(), prefix, filter)
+        // I'm really sorry
+        // We need to set the gitInfo again because if we call this with a custom tag prefix, it will not update the
+        // originally set gitInfo. so this way, we can use the newly-set gitInfo after this method wherever else we
+        // might need it
+        // version = gradleutils.getTagOffsetVersion('different-tag-prefix')
+        var newGitInfo = prefix ? GradleUtils.gitInfoCheckSubproject(project, this.defaultFilter, filter) : GradleUtils.gitInfoCheckSubproject(project, this.defaultFilter, this.defaultFilter, filter);
```

Comment:
This whole change confuses me, why do we need to set the global git state? What is the difference between the prefix and not you pass in defaultFilter twice

Reply:
It's a cope because I have to keep all the existing `gitInfo()` methods in order to not break anything. And honestly I can't be fucked to clean up the rest of this project.

We need to set the global git state again because this method asks to get a different tag prefix or filter, so if we later try to do something like `gradleutils.gitInfo.tag`, it might not give the right one since we specified a different tag in this method. This is done with the intention that `version = getTagOffsetVersion()` is one of the first things that is called in the project's buildscript.